course_info.py

```diff
@@ -0,0 +1,123 @@
+import patterns as pt
+import itertools as it
+
+class Prereq:
```

Comment:
Is the plan here to include all course info into the Prereq class?? If so we should probably change the name of the class to something like `Course_Info`. If not maybe change the name of the file to `prereq.py` or if we want to mirror the naming conventions in the other repo `class-prereq.py`.

Reply:
So the plan was actually to create one or two more classes, `CourseTable`, and maybe `Course` (to be elements of `CourseTable`) which would allow for easy querying against courses.  Then, Prereq would be an attribute in each course (or, as it were, `Course`).  But, I also don't mind renaming this file and giving each class its own file, so I'll leave that up to you.

Reply:
Ok I understand why Prereq is alone now. In Java you need to keep all classes in their own files and a lot of people keep that pattern in other languages as well, with that said I've never been that big of a fan of that pattern, I think related classes can totally stay in the same file.

So all that to say now I understand what you're doing and I'm totally on board.